packages/harpguru-core/src/components/harp-cell/hooks/use-tap-rerender-logic/use-tap-rerender-logic.ts

```diff
@@ -30,21 +31,37 @@ export const useTapRerenderLogic = (
   const [cellState, setCellState] = React.useState(initialCellState)
   const addBufferedActivityToggle = useAddBufferedActivityToggle()
 
-  const tapHandler = (nativeEvent: GestureHandlerStateChangeNativeEvent) => {
+  const onStartSetCellStateWrapper = () => {
+    setCellState(isGloballyActive ? CellStates.TappedOff : CellStates.TappedOn)
+  }
+  const onCancelSetCellStateWrapper = () => {
+    setCellState(isLocallyActive ? CellStates.On : CellStates.Off)
+  }
+  const addBufferedActivityToggleWrapper = () => {
+    // TODO: Is it sufficient to only check for
+    // this value here? ie was this check only
+    // to apease the type system to stop the
+    // next command presenting an error, or does
+    // it actually present some safety? I don't
+    // think it's possible to actually have an
+    // undefined cell run any of this.
```

Comment:
TODO here

Reply:
Even if this is just for type safety I don't have an alternative to leaving it in, so this is a redundant question.

Reply:
Removed